ccaas_builder/cmd/detect/main.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var logger = log.New(os.Stderr, "", 0)
+
+func main() {
+	logger.Println("::Detect")
+
+	if err := run(); err != nil {
+		logger.Printf("::Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	logger.Printf("::Type detected as ccaas")
+
+}
+
+type chaincodeMetadata struct {
+	Type string `json:"type"`
+}
```

Comment:
Type can be lowercase (not exported) as well.

Reply:
Used to unmarshall JSON data into so the elements need to be exported